Changed Font to Inter and added Typescript

wbridegroom opened the pull request:
Gitlab Issue: https://gitlab.mx.com/mx/experiences/money-experience-web/mew/issues/-/issues/520

Commit:
Changed Font to Inter and added Typescript

jgerlach commented on src/core.ts:
```diff
@@ -73,12 +91,12 @@ export default {
-    Semibold: 'ProximaNovaSemibold, Helvetica, Arial, sans-serif',
-    Bold: 'ProximaNovaBold, Helvetica, Arial, sans-serif',
-  },
+    Regular: 'Inter, Helvetica, Arial, sans-serif',
```
Do we want to match the old naming pattern so this is `InterRegular`, `InterSemibold`, `InterBold`?

wbridegroom replied:
I was originally planning to get this route, but it looks like `Inter` doesn't use different names like that. It's simply `Inter` with different weights. Unless I'm missing something.

jgerlach replied:
So, the name itself is actually dependent on how you import the font into your app. Here is an example of how we are currently importing fonts

```
@font-face {
  font-family: 'ProximaNovaSemibold';
  src: url('https://s3.amazonaws.com/path-to-font/fonts/proxima-nova/semibold.eot');
  font-weight: 600;
  font-style: normal;
}
```

You can see that `font-family` is specified as `ProximaNovaSemibold`, but the font file itself isn't named that.

wbridegroom replied:
Changed Inter to match the old naming pattern.

jgerlach commented on src/core.ts:
```diff
@@ -87,43 +105,43 @@ export default {
-  },
+  } as FontWeightType,
   LetterSpacing: {
     Uppercase: 0.48,
```
I'm wondering if we should add the new letter spacing values here instead of just in the letterSpacing file.

jgerlach replied:
Looks like you added them in the `TokenType` file, but not here.

wbridegroom replied:
Merged into core and removed the separate file.

jgerlach commented on src/TokenType.ts:
```diff
@@ -0,0 +1,475 @@
+export interface BackgroundColorType {
```
NIT: I think it would make more sense to call this file `TokenTypes.ts`

wbridegroom replied:
Done

jgerlach commented on src/index.ts:
```diff
@@ -0,0 +1,69 @@
+}
+
+const getSectionTokens = (theme = themes.LIGHT, base = core) => ({
+  BackgroundColor: theme === themes.LIGHT ? backgroundColor.light(base) : backgroundColor.dark(base),
```
Why did you change these to a ternary? Is that because of the typescript change?

wbridegroom replied:
Yes, this is a Typescript issue. It doesn't allow indexing like that.

jgerlach commented on src/mods.ts:
```diff
@@ -1,15 +1,10 @@
-  x2: 0,
-  y2: 0.995,
-})
+const updatedEasing = updateValues(core.Easing, 'cubic-bezier(0.475, 0.425, 0, 0.995)')
```
Is this different then the code on the left? I didn't write it, so I'm actually not sure what it is doing.

wbridegroom replied:
TBH, I'm also not sure what the purpose of this is, because the values provided are identical to the default values.  This was the only way I could get it to work. Out of all the changes I made this is the only one I'm not 100% certain about.

jgerlach replied:
If we notice any issues after we publish we will have to figure out how to fix it.

jgerlach commented on src/tokens/fontFamily.js:
```diff
@@ -1,14 +0,0 @@
-const light = (core) => ({
```
It does look like we aren't using this anywhere. Nice catch.

jgerlach commented on src/tokens/letterSpacing.js:
```diff
@@ -1,14 +0,0 @@
-const light = (core) => ({
```
It looks like the other files were renamed, but this one is showing as deleted. Is that just a git code diff bug?

wbridegroom replied:
I think this was just a diff bug, but I'm removing the TS version anyway now since it's no longer needed.

jgerlach commented on src/utils.ts:
```diff
@@ -98,15 +98,15 @@ export const splitTokens = (tokens, core, leaf) => {
   const tokenList = expandTokens(tokens)
   tokenList.map((token) => {
-    const type = getTokenType(token, core, leaf)
+    const tokenType = getTokenType(token, core, leaf)
```
I wonder if now that we have types for the tokens if this function could be confusing since it's not actually getting TypeScript types

wbridegroom replied:
I agree, but I was trying to avoid making too many changes. Also, I couldn't think of a better name.

smstamm commented on src/core.ts:
```diff
@@ -73,12 +91,12 @@ export default {
-  },
+    Regular: 'Inter, Helvetica, Arial, sans-serif',
+    Semibold: 'Inter, Helvetica, Arial, sans-serif',
+    Bold: 'Inter, Helvetica, Arial, sans-serif',
```
If `Semibold` and `Bold` will be the same as `Regular`, should we just remove these as options and allow them to be controlled in `FontWeight`?

wbridegroom replied:
Long term yes, however, in order to keep this backwards compatible, I think we need to leave it for now.

wbridegroom replied:
Changed Inter to match the old naming pattern, so now we have `InterRegular`, `InterSemiBold` and `InterBold`

Commit:
Requested changes

- Renamed TokenType to TokenTypes
- Changed Inter font to InterRegular, InterSemibold and InterBold
- Merged letterSpacing into Core

smstamm reviewed (approved):
It looks like you addressed all feedback. Looks good to me! Nice work!

Commit:
Changed to major package bump

jgerlach commented on CHANGELOG.md:
```diff
@@ -7,6 +7,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ---
 
+## [11.0.0] - 08-04-2023
```
Thanks for updating this.

jgerlach reviewed (approved):
Great job with the updates and moving to TypeScript
